lte/gateway/python/magma/pipelined/gtp_stats_collector.py

```diff
@@ -0,0 +1,138 @@
+"""
+Copyright 2020 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import logging
+
+import asyncio
+from typing import List
+
+import re
+from collections import namedtuple
+
+from magma.common.job import Job
+from magma.magmad.check import subprocess_workflow
+
+from magma.pipelined.metrics import GTP_PORT_USER_PLANE_DL_BYTES, \
+    GTP_PORT_USER_PLANE_UL_BYTES
+
+OVSDBDumpCommandParams = namedtuple('OVSDBCommandParams', ['table', 'columns'])
+ParsedInterfaceStats = namedtuple('ParsedInterfaceStats', [
+    'Interface',
+    'rx_bytes',
+    'tx_bytes',
+    'remote_ip',
+])
+OVSDBCommandResult = namedtuple('OVSDBCommandResult', ['out', 'err'])
+
+interface_tx_rx_stats_re = re.compile(
+    r'"(?P<Interface>\w+)"(.*)(remote_ip="(?P<remote_ip>.*)"(.*))?rx_bytes=(?P<rx_bytes>\d).+tx_bytes=(?P<tx_bytes>\d)')
+
+MIN_OVSDB_DUMP_POLLING_INTERVAL = 60
+GTP_IP_INTERFACE_PREFIX = 'g_'
+GTP_INTERFACE_PREFIX = 'gtp0'
+
+
+class GTPStatsCollector(Job):
+    def __init__(self, polling_interval: int, service_loop):
+        self._polling_interval = max(polling_interval,
+                                     MIN_OVSDB_DUMP_POLLING_INTERVAL)
+        super().__init__(interval=self._polling_interval, loop=service_loop)
+        self._loop = service_loop
+
+    @asyncio.coroutine
+    def _ovsdb_dump_async(self, table: str, columns: List[str]):
```

Comment:
async def?

Reply:
I tried async def but the issue is that  the subprocess_workflow call complains about the notation with `yield from`:

```
"/home/vagrant/magma/orc8r/gateway/python/magma/magmad/check/subprocess_workflow.py", line 78, in exec_and_parse_subprocesses_async
Oct 06 08:19:56 magma-dev pipelined[28161]:     subprocs = yield from asyncio.gather(*futures)
Oct 06 08:19:56 magma-dev pipelined[28161]:   File "/usr/lib/python3.5/asyncio/futures.py", line 381, in __iter__
Oct 06 08:19:56 magma-dev pipelined[28161]:     assert self.done(), "yield from wasn't used with future"
Oct 06 08:19:56 magma-dev pipelined[28161]: AssertionError: yield from wasn't used with future
```